### ISSUE TITLE
Failure to segment cat-based fusion

### ISSUE THREAD

**Issue by jacobhinkle** [NVIDIA/Fuser#1553]
```python
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_fusion_id1(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, False, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 1, 2, 0])
    T1 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, False, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 1, 2, 0])
    T2 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, False, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 1, 2, 0])
    T3 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[None, None, True, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 2, 1, 0])
    T4 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[None, None, True, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 2, 1, 0])
    T5 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, False, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 1, 2, 0])
    T6 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, False, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 1, 2, 0])
    T7 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, False, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 1, 2, 0])
    T8 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[None, None, True, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 2, 1, 0])
    T9 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[None, None, True, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 2, 1, 0])
    T10 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, False, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 1, 2, 0])
    T11 = fd.define_tensor(shape=[-1, -1, -1, -1], contiguity=[True, True, False, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 1, 2, 0])
    T12 = fd.ops.neg(T2)
    T13 = fd.ops.cat([T12, T1], dim=-1)
    T14 = fd.ops.mul(T0, T3)
    T15 = fd.ops.mul(T13, T4)
    T16 = fd.ops.add(T14, T15)
    T17 = fd.ops.neg(T7)
    T18 = fd.ops.cat([T17, T6], dim=-1)
    T19 = fd.ops.mul(T5, T8)
    T20 = fd.ops.mul(T18, T9)
    T21 = fd.ops.add(T19, T20)
    T22 = fd.ops.cat([T16, T10], dim=-1)
    T23 = fd.ops.cat([T21, T11], dim=-1)
    S24 = fd.define_scalar(0.500000, dtype=DataType.Double)
    T25 = fd.ops.mul(T22, S24)
    T26 = fd.ops.permute(T23, dims=[0, 1, 3, 2])
    S27 = fd.define_scalar(0.500000, dtype=DataType.Double)
    T28 = fd.ops.mul(T26, S27)
    fd.add_output(T13)
    fd.add_output(T18)
    fd.add_output(T25)
    fd.add_output(T28)

with FusionDefinition() as fd:
    nvfuser_fusion_id1(fd)

inputs = [
    torch.randn((4768,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 16), (960, 48, 192, 1)),
    torch.randn((4760,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 8), (960, 48, 192, 1)),
    torch.randn((4760,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 8), (960, 48, 192, 1)),
    torch.randn((80,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 16), (0, 0, 16, 1)),
    torch.randn((80,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 16), (0, 0, 16, 1)),
    torch.randn((4768,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 16), (960, 48, 192, 1)),
    torch.randn((4760,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 8), (960, 48, 192, 1)),
    torch.randn((4760,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 8), (960, 48, 192, 1)),
    torch.randn((80,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 16), (0, 0, 16, 1)),
    torch.randn((80,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 16), (0, 0, 16, 1)),
    torch.randn((0,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 0), (960, 48, 192, 1)),
    torch.randn((0,), dtype=torch.float32, device='cuda:0').as_strided((5, 4, 5, 0), (960, 48, 192, 1)),
]
fd.execute(inputs)
```
Fails with
> RuntimeError: h.has_value() INTERNAL ASSERT FAILED at "/opt/pytorch/nvfuser/csrc/fusion_segmenter.cpp":3556, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/Fuser/issues. Can not find a scheduler to schedule fusion segment

**Comment by jacobhinkle** [comment]
Possible dupe of #1541.

**Comment by tfogal** [comment]
Hitting this when trying to force llama2 into bfloat16:

```diff
@@ -157,7 +157,7 @@ if init_from == "scratch":
     # init a new model from scratch
     #print("Initializing a new model from scratch")
     gptconf = ModelArgs(**model_args)
-    model = Transformer(gptconf)
+    model = Transformer(gptconf).to(device=device_type, dtype=torch.bfloat16)
 elif init_from == "resume":
     print(f"Resuming training from {out_dir}")
     # resume training from a checkpoint.
```

**Comment by wujingyue** [comment]
Fixed by #1825

**Comment by wujingyue** [comment]
> Also hitting this when trying to force llama2 into bfloat16:

Just a heads-up, while I'm marking this as fixed, you may still hit a (different) bug: https://github.com/NVIDIA/Fuser/issues/1557
